Neos.ContentRepository.LegacyNodeMigration/Classes/Command/SiteCommandController.php

```diff
@@ -18,50 +18,39 @@
 use Doctrine\DBAL\DriverManager;
 use Doctrine\DBAL\Exception as DBALException;
 use Doctrine\DBAL\Exception\ConnectionException;
-use Neos\ContentRepository\Core\Projection\CatchUpOptions;
-use Neos\ContentRepository\Core\SharedModel\Workspace\ContentStreamId;
+use Neos\ContentRepository\Core\SharedModel\ContentRepository\ContentRepositoryId;
+use Neos\ContentRepository\Export\Severity;
+use Neos\ContentRepository\LegacyNodeMigration\LegacyExportServiceFactory;
 use Neos\ContentRepository\LegacyNodeMigration\LegacyMigrationService;
 use Neos\ContentRepository\LegacyNodeMigration\LegacyMigrationServiceFactory;
 use Neos\ContentRepository\LegacyNodeMigration\RootNodeTypeMapping;
 use Neos\ContentRepositoryRegistry\ContentRepositoryRegistry;
-use Neos\ContentRepositoryRegistry\Factory\EventStore\DoctrineEventStoreFactory;
-use Neos\ContentRepositoryRegistry\Service\ProjectionReplayServiceFactory;
 use Neos\Flow\Cli\CommandController;
-use Neos\Flow\Persistence\PersistenceManagerInterface;
 use Neos\Flow\Property\PropertyMapper;
-use Neos\Flow\ResourceManagement\ResourceManager;
-use Neos\Flow\ResourceManagement\ResourceRepository;
 use Neos\Flow\Utility\Environment;
-use Neos\Media\Domain\Repository\AssetRepository;
-use Neos\Neos\Domain\Model\Site;
-use Neos\Neos\Domain\Repository\SiteRepository;
+use Neos\Neos\Domain\Service\SiteImportService;
+use Neos\Utility\Files;
 use Neos\Neos\Domain\Service\NodeTypeNameFactory;
 
-class CrCommandController extends CommandController
+class SiteCommandController extends CommandController
 {
     public function __construct(
         private readonly Connection $connection,
         private readonly Environment $environment,
-        private readonly PersistenceManagerInterface $persistenceManager,
-        private readonly AssetRepository $assetRepository,
-        private readonly ResourceRepository $resourceRepository,
-        private readonly ResourceManager $resourceManager,
         private readonly PropertyMapper $propertyMapper,
         private readonly ContentRepositoryRegistry $contentRepositoryRegistry,
-        private readonly SiteRepository $siteRepository,
-        private readonly ProjectionReplayServiceFactory $projectionReplayServiceFactory,
+        private readonly SiteImportService $siteImportService,
     ) {
         parent::__construct();
     }
 
     /**
      * Migrate from the Legacy CR
      *
-     * @param bool $verbose If set, all notices will be rendered
      * @param string|null $config JSON encoded configuration, for example '{"dbal": {"dbname": "some-other-db"}, "resourcesPath": "/some/absolute/path", "rootNodes": {"/sites": "Neos.Neos:Sites", "/other": "My.Package:SomeOtherRoot"}}'
      * @throws \Exception
      */
-    public function migrateLegacyDataCommand(bool $verbose = false, string $config = null): void
+    public function migrateLegacyDataCommand(string $contentRepository = 'default', bool $verbose = false, string $config = null): void
```

Comment:
i wonder is it of any help to do this in one step and have no in between results?

Currently if anything fails in the `importFromPath` we dont even unlink the `$temporaryFilePath` and it would be frustrating i would have nothing agains using the new `exportLegacyData` only like:

```
flow site:exportLegacyDataCommand --path /temp/but/the/user/sees/it
flow site:importAll --path /temp/but/the/user/sees/it
```

Otherwise we would need to check all preconditions already in `importAll` like is doctrine migrated etc ...
I think it would be good to enforce that transparency and make things easier for us (less duplicated code ..., less long running processes where steps could fail ANYWHERE) no?

Reply:
As discussed: I think it's a good idea to separate the two, i.e. instead of migrating directly to export legacy sites into the new import format and then use the same entry point for both ways to _import_ a new site